oci/oci.go

```diff
@@ -71,41 +63,43 @@ type RuntimeBase struct {
 	logSizeMax               int64
 	noPivot                  bool
 	ctrStopTimeout           int64
+
+	runtimeImplList map[string]runtimeImpl
+	runtimeType     string
 }
 
-// RuntimeImpl is an interface used by the caller to interact with the
+// runtimeImpl is an interface used by the caller to interact with the
 // container runtime. The purpose of this interface being to abstract
 // implementations and their associated assumptions regarding the way to
 // interact with containers. This will allow for new implementations of
 // this interface, especially useful for the case of VM based container
 // runtimes. Assumptions based on the fact that a container process runs
-// on the host will be limited to the RuntimeV1 implementation.
-type RuntimeImpl interface {
-	Version() (string, error)
-	CreateContainer(*Container, string) error
-	StartContainer(*Container) error
-	ExecContainer(*Container, []string, io.Reader, io.WriteCloser, io.WriteCloser,
+// on the host will be limited to the RuntimeOCI implementation.
+type runtimeImpl interface {
```

Comment:
Any reason why we made this private? (Sorry some comments may be premature as I make my way through the changes).

Reply:
Since it's internal to the `oci` package, there is no reason to make it public. By default, I always try to keep things private if they don't need to be exported for consumers of this package.